raiden/raiden_service.py

```diff
@@ -366,8 +378,12 @@ def get_new_events(self):
 
         return event_list
 
-    def open(self, token_address, partner_address,
-             settle_timeout=None, reveal_timeout=None):
+    def open(
+            self,
+            token_address,
+            partner_address,
+            settle_timeout=None,
+            reveal_timeout=None):
```

Comment:
Synchronization for these stateful APIs is still a nice to have.

#### stateful functions exposed with the current REST API draft

- get_new_events
- open
- deposit
- close
- settle

#### stateful functions not exposed with the current REST API draft

- transfer
- transfer_and_wait
- transfer_async

*Note*: Even if these functions are exposed in the future, we don't need to control the concurrent usage.

#### semantics for the concurrency control

- linearize the calls:
  - synchronized for the same cursor
    - get_new_events
- prohibit concurrent calls
  - channel manager with the same partner address
    - open
  - on the channel
    - deposit, close, and settle

*Reminder*: I'm not implying that these functions are not safe for concurrent usage, they must be safe to use concurrently, but that it's nice to avoid execution of unnecessary transactions.

Maybe we should move this to a new issue? Since this is not exclusive to the rest api and it's not a functionality problem, but a protection, so to speak, to avoid burning gas.